service/frontend/workflow_handler.go

```diff
@@ -5180,6 +5202,569 @@ func (wh *WorkflowHandler) RespondNexusTaskFailed(ctx context.Context, request *
+	endpointEntry, err := wh.endpointRegistry.GetByName(ctx, ns.ID(), endpoint)
+	if err != nil {
+		return nil, err
+	}
+	nc, err := wh.nexusClientProvider(ctx, ns.ID().String(), endpointEntry, request.Service)
+	if err != nil {
+		return nil, err
+	}
+
+	rawResult, err := nc.StartOperation(ctx, request.Operation, request.Payload, nexus.StartOperationOptions{
+		RequestID:      request.RequestId,
+		Header:         request.Header,
+		CallbackURL:    request.Callback,
+		CallbackHeader: request.CallbackHeader,
+		Links:          parseLinks(request.Links, wh.logger),
+	})
+
+	if err != nil {
+		var handlerErr *nexus.HandlerError
+		if errors.As(err, &handlerErr) {
+			return &workflowservice.StartNexusOperationResponse{
+				Variant: &workflowservice.StartNexusOperationResponse_HandlerError{
+					HandlerError: commonnexus.NexusHandlerErrorToProtoHandlerError(handlerErr),
+				},
+			}, nil
+		}
+		var opFailedErr *nexus.OperationError
+		if errors.As(err, &opFailedErr) {
+			return &workflowservice.StartNexusOperationResponse{
+				Variant: &workflowservice.StartNexusOperationResponse_Unsuccessful_{
+					Unsuccessful: &workflowservice.StartNexusOperationResponse_Unsuccessful{
+						OperationError: commonnexus.NexusOpErrorToProtoOpError(opFailedErr),
+					},
+				},
+			}, nil
+		}
+		wh.logger.Error("received unexpected error for start Nexus operation HTTP request", tag.Operation("StartNexusOperation"), tag.WorkflowNamespace(ns.Name().String()), tag.Error(err))
+		return nil, serviceerror.NewInternal("internal error")
+	}
+
+	links := make([]*nexuspb.Link, len(rawResult.Links))
+	for i, l := range rawResult.Links {
+		links[i] = &nexuspb.Link{
+			Type: l.Type,
+			Url:  l.URL.String(),
+		}
+	}
+	if rawResult.Pending != nil {
+		return &workflowservice.StartNexusOperationResponse{
+			Links: links,
+			Variant: &workflowservice.StartNexusOperationResponse_AsyncSuccess{
+				AsyncSuccess: &workflowservice.StartNexusOperationResponse_Async{
+					OperationToken: rawResult.Pending.Token,
+				},
+			},
+		}, nil
+	}
+
+	var res *commonpb.Payload
+	if rawResult.Successful != nil {
+		err := rawResult.Successful.Consume(&res)
+		if err != nil {
+			return nil, serviceerror.NewInternalf("failed to deserialize Nexus operation result: %s", err.Error())
+		}
+	}
+	return &workflowservice.StartNexusOperationResponse{
+		Links: links,
+		Variant: &workflowservice.StartNexusOperationResponse_SyncSuccess{
+			SyncSuccess: &workflowservice.StartNexusOperationResponse_Sync{
+				Result: res,
+			},
+		},
+	}, nil
+}
+
+func (wh *WorkflowHandler) dispatchStartNexusOperation(ctx context.Context, ns *namespace.Namespace, taskQueue string, request *workflowservice.StartNexusOperationRequest) (*workflowservice.StartNexusOperationResponse, error) {
+	sizeLimitError := wh.config.BlobSizeLimitError(ns.Name().String())
+	sizeLimitWarn := wh.config.BlobSizeLimitWarn(ns.Name().String())
+	if err := common.CheckEventBlobSizeLimit(
```

Comment:
This should apply to both the HTTP and matching call. No need to forward a request that we know we can't accept.